python/psautohint/ufoFont.py

```diff
@@ -897,15 +900,22 @@ def getHintSet(self):
 
         if len(self.hList) > 0 or len(self.hstem3List):
             hintset[STEMS_NAME].extend(
-                    makeHintSet(self.hList, self.hstem3List, isH=True))
+                makeHintSet(self.hList, self.hstem3List, isH=True))
 
         if len(self.vList) > 0 or len(self.vstem3List):
             hintset[STEMS_NAME].extend(
-                    makeHintSet(self.vList, self.vstem3List, isH=False))
+                makeHintSet(self.vList, self.vstem3List, isH=False))
 
         return hintset
 
 
+def norm_float(value):
+    """Converts a float (whose decimal part is zero) to integer"""
+    if isinstance(value, float) and value % 1 == 0:
```

Comment:
`float.is_integer` is more idiomatic (not sure if faster)
https://docs.python.org/3/library/stdtypes.html#float.is_integer

Reply:
```
~$ python3 -m timeit -s '(-2.0).is_integer()'
50000000 loops, best of 5: 7.93 nsec per loop
--------------------------------------------------------------------------------------------
~$ python3 -m timeit -s 'isinstance(-2.0, float) and -2.0 % 1 == 0'
50000000 loops, best of 5: 8.02 nsec per loop
```
`float.is_integer` is faster. I'll change it.

Reply:
interesting. very small margin. I prefer is_interger mostly for legibility

Reply:
However, the incoming values are integers sometimes, and `(-2).is_integer()` yields an `AttributeError`, which means we need to have an extra check. Need to think about this some more.

Reply:
yeah you still need to check isinstance(v, float) of course

Reply:
Right. So a better time comparison is this:
```
~$ python3 -m timeit -s 'isinstance(-2.0, float) and -2.0 % 1 == 0'
50000000 loops, best of 5: 8.03 nsec per loop
----------------------------------------------------------------------------------
~$ python3 -m timeit -s 'isinstance(-2.0, float) and (-2.0).is_integer()'
50000000 loops, best of 5: 8.02 nsec per loop
```